java/ql/src/Telemetry/APIUsage.qll

```diff
@@ -0,0 +1,44 @@
+import java
+private import semmle.code.java.dataflow.FlowSteps
+private import semmle.code.java.dataflow.ExternalFlow
+
+string jarName(CompilationUnit cu) {
+  result = cu.getParentContainer().toString().regexpCapture(".*/(.*\\.jar)/?.*", 1)
+}
+
+predicate isJavaRuntime(Callable call) {
+  jarName(call.getCompilationUnit()) = "rt.jar" or
+  call.getCompilationUnit().getParentContainer().toString().substring(0, 14) = "/modules/java."
+}
+
+// TODO Is this heuristic too broad?
+predicate isInterestingAPI(Callable call) {
+  call.getNumberOfParameters() > 0 and
+  not (
+    call.getReturnType() instanceof VoidType or
+    call.getReturnType() instanceof PrimitiveType or
+    call.getReturnType() instanceof BoxedType
+  )
```

Comment:
Minor performance note: Complex negations mean that the negation body must be computed in order to do set subtraction, whereas simple negations that merely check membership in an existing relation will be compiled to efficient anti-joins. This is a case that can be rewritten to a sequence of anti-joins by distributing the negation and pulling the `getReturnType()` from the negation (which depends on the predicate being functional to preserve the semantics). This isn't likely to make any big difference here, but it's good to get in the habit of writing code that compiles well.
```suggestion
  exists(Type retType | retType = call.getReturnType() |
    not retType instanceof VoidType and
    not retType instanceof PrimitiveType and
    not retType instanceof BoxedType
  )
```

Reply:
Thanks for the detailed explanation. That makes sense and I'll try to do it this way in the future. Applied manually